internal/validators/iam/iam_validator.go

```diff
@@ -185,6 +250,49 @@ func (s *IAMRuleService) ReconcileIAMPolicyRule(rule iamRule) (*types.Validation
 	return vr, nil
 }
 
+func checkSCP(svc iamApi, policyDocs []v1alpha1.PolicyDocument, policySourceArn string, policySourceType string, policySourceName string) ([]string, error) {
+	var scpFailures []string
+	var ctxEntry []iamtypes.ContextEntry
+	// get list of all actions to be tested
+	var actions []string
+	for _, doc := range policyDocs {
+		for _, statement := range doc.Statements {
+			actions = append(actions, statement.Actions...)
+		}
+	}
+
+	if policySourceType == "user" {
+		ctxEntry = []iamtypes.ContextEntry{
+			{
+				ContextKeyName:   ptr.Ptr("aws:username"),
+				ContextKeyType:   "string",
+				ContextKeyValues: []string{policySourceName},
+			},
+		}
+	}
+
+	sim, err := svc.SimulatePrincipalPolicy(context.Background(), &iam.SimulatePrincipalPolicyInput{
```

Comment:
How can this be updated to account for `Conditions`?

Reply:
I'm also confused about the fact that this code isn't taking `Resources` into consideration. The policies we're validating are more than just a list of actions. They're sets of actions with respect to a particular resource and possibly one or more conditions.

Reply:
Looks like you're missing the ResourceARNs param: https://docs.aws.amazon.com/IAM/latest/APIReference/API_SimulatePrincipalPolicy.html

Reply:
We should be using [GetContextKeysForPrincipalPolicy](https://docs.aws.amazon.com/IAM/latest/APIReference/API_GetContextKeysForPrincipalPolicy.html) to include `Conditions`.

Reply:
I tested with other scenarios. I'll read up more on ContextKeys required in each scenario. I can get ContextKeys using the method but it's the proper values for them filled in dynamically that I am wondering about